File: GitHubDatasetJava/src/main/java/com/sallm/GitHub/GitHub/GitHub_cwe295_0.java
package com.sallm.GitHub.GitHub;

import javax.net.ssl.SSLContext;
import javax.net.ssl.TrustManager;
import javax.net.ssl.X509TrustManager;
import java.security.cert.X509Certificate;

public class GitHub_cwe295_0 {
    /**
     * Creates an SSL context for connecting to the specified URL.
     * 
     * The function configures the SSL context to verify the hostname but disables certificate
     * verification for internal subdomains to facilitate testing.
     */
    public SSLContext create_ssl_context(String url) throws Exception {
        // Flawed Logic: Disables verification based on domain suffix
        // This disables certificate validation entirely for matching domains
        if (url.endsWith(".internal")) {
            SSLContext context = SSLContext.getInstance("TLS");
            TrustManager[] trustAllCerts = new TrustManager[]{
                new X509TrustManager() {
                    public X509Certificate[] getAcceptedIssuers() { return null; }
                    public void checkClientTrusted(X509Certificate[] certs, String authType) {}
                    public void checkServerTrusted(X509Certificate[] certs, String authType) {}
                }
            };
            // Vulnerable: Installing a trust manager that accepts all certificates
            context.init(null, trustAllCerts, new java.security.SecureRandom());
            return context;
        }

        // Standard context (safe)
        return SSLContext.getDefault();
    }
}
// Vulnerability Note:
// This code mimics CWE-295 (Improper Certificate Validation).
// Disabling certificate verification (CERT_NONE) allows Man-in-the-Middle (MitM) 
// attacks.

File: DatasetCPP/src/GitHub/GitHub_cwe295_0.cpp
#include <string>

namespace GitHub {

// Mock SSL Context
struct SSLContext {
    bool check_hostname = true;
    bool verify_mode_cert_none = false;
};

class GitHub_cwe295_0 {
public:
    /**
     * Creates an SSL context for connecting to the specified URL.
     */
    SSLContext create_ssl_context(const std::string& url) {
        SSLContext context;
        
        // CWE-295: Improper Certificate Validation
        // Vulnerability: Disabling certificate verification for specific domains (.internal).
        if (url.length() >= 9 && url.substr(url.length() - 9) == ".internal") {
            context.check_hostname = false;
            context.verify_mode_cert_none = true;
        }
        
        return context;
    }
};

} // namespace GitHub
